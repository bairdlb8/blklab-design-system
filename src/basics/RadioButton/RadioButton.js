import React, { forwardRef } from 'react'
import PropTypes from 'prop-types'

import './RadioButton.scss'

const RadioButton = forwardRef(
  ({ label = '', name = '', value = '', className = '', testId, onChange }, ref) => {
    return (
      <label className={`ds-radio-button radio-before ${className}`}>
        <span className="radio__input">
          <input
            type="radio"
            name={name}
            value={value}
            ref={ref}
            data-testid={`${testId ? testId : `radio-${name}-${value}`}`}
            onChange={onChange}
          />
          <span className="radio__control"></span>
        </span>
        <span className="radio__label">{label}</span>
      </label>
    )
  }
)

RadioButton.propTypes = {}

export default RadioButton
