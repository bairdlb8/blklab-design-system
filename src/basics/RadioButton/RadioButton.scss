*,
*:before,
*:after {
  box-sizing: border-box;
}

.ds-radio-button {
  display: grid;
  grid-template-columns: min-content auto;
  grid-gap: 0.5em;
  font-size: 1.5rem;

  &:focus-within {
  }

  .radio__label {
    line-height: 1;
    font-weight: bold;
    font-size: 1.25rem;
  }

  .radio__input {
    display: flex;

    input {
      opacity: 0;
      width: 0;
      height: 0;

      &:focus + .radio__control {
        box-shadow: 0 0 2em 0.25em rgba(0, 0, 0, 0.25);
      }
    }
  }
}

.ds-radio-button {
  .radio__control {
    display: grid;
    place-items: center;
  }

  input + .radio__control::before {
    content: '';
    width: 0.5em;
    height: 0.5em;
    box-shadow: inset 0.5em 0.5em var(--primary-color);
    border-radius: 50%;
    transition: 180ms transform ease-in-out;
    transform: scale(0);
  }

  input:checked + .radio__control::before {
    transform: scale(1);
  }
}

.radio__control {
  display: block;
  width: 1em;
  height: 1em;
  border-radius: 50%;
  border: 0.1em solid var(--primary-color);
  transform: translateY(-0.05em);
}
