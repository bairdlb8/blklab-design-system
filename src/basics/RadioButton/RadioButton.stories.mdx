import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks'

import RadioButton from './RadioButton'

<Meta title="Basics / RadioButton" component={RadioButton} />

# RadioButton

## Add some text here

export const Template = (args) => {
  return <RadioButton {...args} />
}

<Preview withToolbar>
  <Story name="Primary" args={{ label: 'Test' }}>
    {Template.bind({})}
  </Story>
</Preview>

<Props of={RadioButton} />
