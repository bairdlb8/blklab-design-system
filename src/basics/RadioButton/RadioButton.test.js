import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { toHaveNoViolations } from 'jest-axe'
import RadioButton from './RadioButton'

expect.extend(toHaveNoViolations)

describe('<RadioButton />', () => {

  describe('accessibility', () => {
    it('should not fail any accessibility tests', async () => {
      const { container } = render(<RadioButton />)
      expect(await axe(container)).toHaveNoViolations()
    })
  })

  it('should take a snapshot', () => {
    const { asFragment } = render(<RadioButton />)

    expect(asFragment(<RadioButton />)).toMatchSnapshot()
  })

})
