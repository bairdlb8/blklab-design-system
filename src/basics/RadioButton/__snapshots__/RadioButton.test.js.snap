// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    RadioButton
  </div>
</DocumentFragment>
`;
